Ice/Settings/SettingsPanes/GeneralSettingsPane.swift

```diff
@@ -12,54 +12,55 @@ struct GeneralSettingsPane: View {
     @State private var isImportingCustomIceIcon = false
     @State private var isPresentingError = false
     @State private var presentedError: AnyLocalizedError?
+    @State private var isApplyingOffset = false
+    @State private var tempItemSpacingOffset: CGFloat = 0 // Temporary state for the slider
 
     private var manager: GeneralSettingsManager {
         appState.settingsManager.generalSettingsManager
     }
 
     private var itemSpacingOffset: LocalizedStringKey {
-        if manager.itemSpacingOffset == -16 {
-            LocalizedStringKey("none")
-        } else if manager.itemSpacingOffset == 0 {
-            LocalizedStringKey("default")
-        } else if manager.itemSpacingOffset == 16 {
-            LocalizedStringKey("max")
-        } else {
-            LocalizedStringKey(manager.itemSpacingOffset.formatted())
+        localizedOffsetString(for: manager.itemSpacingOffset)
+    }
+
+    private func localizedOffsetString(for offset: CGFloat) -> LocalizedStringKey {
+        switch offset {
+        case -16:
+            return LocalizedStringKey("none")
+        case 0:
+            return LocalizedStringKey("default")
+        case 16:
+            return LocalizedStringKey("max")
+        default:
+            return LocalizedStringKey(offset.formatted())
         }
     }
 
     private var rehideInterval: LocalizedStringKey {
         let formatted = manager.rehideInterval.formatted()
-        return if manager.rehideInterval == 1 {
-            LocalizedStringKey(formatted + " second")
+        if manager.rehideInterval == 1 {
+            return LocalizedStringKey(formatted + " second")
         } else {
-            LocalizedStringKey(formatted + " seconds")
+            return LocalizedStringKey(formatted + " seconds")
         }
     }
 
+    private var hasSliderValueChanged: Bool {
+        tempItemSpacingOffset != manager.itemSpacingOffset
+    }
+
+    private var isActualValueDifferentFromDefault: Bool {
+        manager.itemSpacingOffset != 0
+    }
+
     var body: some View {
         Form {
-            Section {
-                launchAtLogin
-            }
-            Section {
-                iceIconOptions
-            }
-            Section {
-                useIceBar
-            }
-            Section {
-                showOnClick
-                showOnHover
-                showOnScroll
-            }
-            Section {
-                autoRehideOptions
-            }
-            Section {
-                spacingOptions
-            }
+            Section { launchAtLogin }
+            Section { iceIconOptions }
+            Section { useIceBar }
+            Section { showOnClick; showOnHover; showOnScroll }
+            Section { autoRehideOptions }
+            Section { spacingOptions }
```

Comment:
Let's keep the original formatting here.

```suggestion
            Section {
                launchAtLogin
            }
            Section {
                iceIconOptions
            }
            Section {
                useIceBar
            }
            Section {
                showOnClick
                showOnHover
                showOnScroll
            }
            Section {
                autoRehideOptions
            }
            Section {
                spacingOptions
            }
```